h2h/production-code/week6/get_data.py

```diff
@@ -4,85 +4,131 @@
-    containing the players on each team, and a record of the team ID of both the
-    winning team and the home team to help encode the dataset. Then, encodes it,
-    placing 1 or -1 appropriately on each player's column. Intended to be used
-    for machine learning.
+    Calls _generate_player_in_game_dataset or reads a dataset that contains a
+    record of games, with columns containing the players on each team, and a
+    record of the team ID of both the winning team and the home team to help
+    encode the dataset. Then, encodes it, placing 1 or -1 appropriately on each
+    player's column.
 
     Args:
-        season_start: First season of data to encode. If you put in 2015, this
-            means it will use the 2015-2016 season.
-        season_end: Last season of data to encode, inclusive. Same format as
-            above.
-        read_filepath: Without this arg, this method will generate a new datset.
-            If given, it will instead read a CSV with the given name.
-        write_filepath: By default, function onloy returns a dataframe. If you
-            want to save it, put a .csv file name in this arg
-        write_filepath_unencoded: If you are generating a raw dataset first,
-            this will save it instead of having to generate it each time.
+        read_unencoded: Without this arg, this method will generate a new
+            dataset. If given, it will instead read a CSV with the given name,
+            and then encode it.
+        write_encoded: By default, function doesn't save the encoded dataset.
+            If you want to save the encoded dataframe, put a .csv file name in
+            this arg.
+        season_start: Used only if you want to call
+            _generate_player_in_game_dataset. See downstream description.
+        season_end: Used only if you want to call
+            _generate_player_in_game_dataset. See downstream description.
+        write_unencoded: Used only if you want to call
+            _generate_player_in_game_dataset. See downstream description.
 
     Returns:
         A dataframe that has columns for every individual player in the raw
         dataset that was either read from CSV or generated, plus a label column
         indicating the outcome of the game. Each row represents a game. A
         player's value in a given row is -1 if they were on the away team, 1 if
-        trhey were on the home team. Label column is 1 for a home win, 0 for an
+        they were on the home team. Label column is 1 for a home win, 0 for an
         away win.
 
     Raises:
         None
     """
-    if read_filepath:
-        raw_dataset = pd.read_csv(read_filepath)
-        #when we read csv, a set will be interpreted as a string so we have to evaluate
+    if read_unencoded: #have unencoded csv, just need to encode
+        raw_dataset = pd.read_csv(read_unencoded)
         raw_dataset["winning_team_players"] = raw_dataset["winning_team_players"].apply(lambda x:literal_eval(x))
```

Comment:
from ast import literal_eval?

Reply:
what does this mean?

Reply:
I don't think you imported / defined literal_eval anywhere

Reply:
it's at the top in the import statements... good eye though, thanks

Reply:
![image](https://user-images.githubusercontent.com/42979139/86177476-bafab300-baf4-11ea-8e4c-22f1dfe5e9e9.png)
Good to know it doesn't show the whole file

Reply:
Gotta click the button :(

Reply:
interesting